espresso/devnet-tests/devnet_tools.go

```diff
@@ -376,6 +377,158 @@ func (d *Devnet) Down() error {
 	return cmd.Run()
 }
 
+type TaggedWriter struct {
+	inner   io.Writer
+	tag     string
+	newline bool
+}
+
+func NewTaggedWriter(tag string, inner io.Writer) *TaggedWriter {
+	return &TaggedWriter{
+		inner:   inner,
+		tag:     tag,
+		newline: true,
+	}
+}
+
+func (w *TaggedWriter) Write(p []byte) (int, error) {
```

Comment:
What does this function do? Where does it write and why?

Reply:
This is the generic `go` io writer interface. This type is just a wrapper around an inner IO stream that prepends output with a tag (e.g. `op-challenger | ...`). I'll add commenets